glfs.c

```diff
@@ -412,7 +412,7 @@ static glfs_t* tcmu_create_glfs_object(char *config, gluster_server **hosts)
 	gluster_cache_refresh(fs, config);
 
  fail:
-	gluster_free_server(&entry);
+	gluster_free_server(hosts);
```

Comment:
Thanks, I think hosts ptr should have always been NULL. I come from glusterfs code where ptrs are always inited to NULL. Prasanna worked mostly with qemu where ptrs inited only when it is needed. So between both of us we managed to not handle this part in glfs_check_config(). Let me send a patch for the same. In all other cases it shouldn't be allocated otherwise it would be a leak. For now I don't see such code paths. Do let me know if I missed something.